src/crimson/os/seastore/segment_manager/zns.cc

```diff
@@ -467,17 +488,16 @@ SegmentManager::read_ertr::future<> ZNSSegmentManager::read(
   size_t len,
   ceph::bufferptr &out)
 {
+  LOG_PREFIX(ZNSSegmentManager::read);
   auto& seg_addr = addr.as_seg_paddr();
   if (seg_addr.get_segment_id().device_segment_id() >= get_num_segments()) {
-    logger().error(
-      "ZNSSegmentManager::read: invalid segment {}",
+    ERROR("invalid segment {}",
       addr);
     return crimson::ct_error::invarg::make();
   }
   
   if (seg_addr.get_segment_off() + len > metadata.zone_size) {
```

Comment:
seg_addr.get_segment_off() + len > metadata.segment_capacity

Reply:
Will keep it at segment size for now, will handle the capacity changes in subsequent patches.